File: scripts/algorithms/Longest Absolute File Path/Longest Absolute File Path.js
function isFile(path) {
    return path.includes('.')
}

var lengthLongestPath = function(input) {
    const segments = input.split('\n');
    
    let max = 0;
    let path = [];
    for (const segment of segments) {
        if (segment.startsWith('\t')) {
            const nesting = segment.match(/\t/g).length;
            
            while (nesting < path.length) {
                path.pop();
            }
            
            path.push(segment.replace(/\t/g, ''))
        } else {
            path = [segment]
        }
        
        if (isFile(path.at(-1))) {
            const filePath = path.join('/');
            if (filePath.length > max) {
                max = filePath.length;
            }
        }
    }
    
    return max;
};

File: scripts/Longest Absolute File Path/Longest Absolute File Path.py
class Solution:
	def lengthLongestPath(self, input: str) -> int:
		if "." not in input:
			return 0

		a=input.split("\n")
		files=[]
		for i in a:
			if "." in i:
				files.append(i)

		final=[]
		for i in range(len(files)):
			file=files[i]
			lvl=file.count("\t")
			idx=a.index(file)-1
			save=[files[i].replace("\t","")]
			for j in range(lvl):
				while a[idx].count("\t")!=lvl-1:
					idx-=1
				lvl=a[idx].count("\t")
				save.append(a[idx].replace("\t",""))
				idx-=1
			final.append(save)

		final=list(map("/".join,final))
		return len(max(final,key=len))